src/domains/wallet/components/VerifyMessage/VerifyMessage.tsx

```diff
@@ -5,37 +5,65 @@ import { Form, FormField, FormHelperText, FormLabel } from "app/components/Form"
 import { Input } from "app/components/Input";
 import { Modal } from "app/components/Modal";
 import { Toggle } from "app/components/Toggle";
+import { useEnvironment } from "app/contexts";
 import React, { useState } from "react";
 import { useForm } from "react-hook-form";
 import { useTranslation } from "react-i18next";
 
 type Props = {
 	onSubmit?: any;
+	onCancel?: any;
 	isOpen: boolean;
 	handleClose?: any;
-	publicKey?: string;
+	signatory?: string;
+	profileId: string;
+	walletPublicKey: string;
 };
 
-export const VerifyMessage = ({ onSubmit, publicKey, isOpen, handleClose }: Props) => {
+export const VerifyMessage = ({
+	profileId,
+	walletPublicKey,
+	onSubmit,
+	onCancel,
+	signatory,
+	isOpen,
+	handleClose,
+}: Props) => {
+	const env = useEnvironment();
 	const form = useForm();
-
 	const { t } = useTranslation();
 
 	const { register } = form;
 	const [verifyAddress, setVerifyAddress] = useState(true);
 
+	const handleSubmit = async () => {
+		let isVerified = false;
+		const formValues = form.getValues();
+		const profile = env?.profiles().findById(profileId);
+		const wallet = profile?.wallets().findByPublicKey(walletPublicKey);
+
+		try {
+			const signedMessage = verifyAddress ? JSON.parse(formValues["signet-message-content"]) : formValues;
```

Comment:
```suggestion
			const signedMessage = verifyAddress ? JSON.parse(formValues["signed-message-content"]) : formValues;
```

Reply:
@goga-m this `signet` instead of `signed` issue is in a few other places, I just didn't want to point out all of them with a comment.